configurer/path_windows.go

```diff
@@ -3,48 +3,84 @@
 package configurer
 
 import (
+	"fmt"
 	"os"
 	"path/filepath"
 	"syscall"
+
+	"github.com/pastelnetwork/pastel-utility/constants"
 )
 
 const (
 	beforeVistaAppDir = "Application Data"
 	sinceVistaAppDir  = "AppData/Roaming"
 )
 
+// GetHomeDir returns the home path for Windows OS.
+func GetHomeDir() (string, error) {
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		return "", err
+	}
+	return homeDir, nil
+}
+
 // DefaultWorkingDir returns the default config path for Windows OS.
-func DefaultWorkingDir() string {
-	homeDir, _ := os.UserHomeDir()
+func DefaultWorkingDir() (string, error) {
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		return "", err
+	}
 	appDir := beforeVistaAppDir
 
 	v, _ := syscall.GetVersion()
 	if v&0xff > 5 {
 		appDir = sinceVistaAppDir
 	}
-	return filepath.Join(homeDir, filepath.FromSlash(appDir), "Pastel")
+
+	return filepath.Join(homeDir, filepath.FromSlash(appDir), "Pastel"), nil
 }
 
 // DefaultZksnarkDir returns the default config path for Windows OS.
-func DefaultZksnarkDir() string {
-	homeDir, _ := os.UserHomeDir()
+func DefaultZksnarkDir() (string, error) {
```

Comment:
There is only a difference of 1 string val in func DefaultZksnarDir() &  DefaultPastelExecutableDir()
The difference is in the last line of both functions. You can easily write one function for it.